// protobuf for dab-tree

syntax = "proto3";
package tree;

enum FeaType {
  CONT = 0;
  DISC = 1;
  RANK = 2;
}

message Value {
  oneof values {
    float v = 1;
    uint64 id = 2;
    int64 n = 3;
  }
}

message RegTree {
  // uint64 max_node_size = 0;
  uint64 id = 1;
  repeated FeaType fea_types = 2 [packed=true];
  repeated uint64 split_fea = 3 [packed=true];
  repeated Value split_value = 4;
  repeated bool is_leaf = 5 [packed=true];
}
